dev/TabView/TabViewItem.cpp

```diff
@@ -97,7 +96,33 @@ void TabViewItem::OnApplyTemplate()
 
     UpdateCloseButton();
     UpdateForeground();
-    UpdateWidthModeVisualState();    
+    UpdateWidthModeVisualState();
+    UpdateTabGeometry();
+}
+
+void TabViewItem::UpdateTabGeometry()
+{
+    auto const templateSettings = winrt::get_self<TabViewItemTemplateSettings>(TabViewTemplateSettings());
+
+    auto const height = ActualHeight();
+    auto const popupRadius = unbox_value<winrt::CornerRadius>(ResourceAccessor::ResourceLookup(*this, box_value(c_overlayCornerRadiusKey)));
+    auto const leftCorner = popupRadius.TopLeft;
+    auto const rightCorner = popupRadius.TopRight;
+
+    // Assumes 4px curving-out corners, which are hardcoded in the markup
+    auto data = L"<Geometry xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'>F1 M0,%f  a 4,4 0 0 0 4,-4  L 4,%f  a %f,%f 0 0 1 %f,-%f  l %f,0  a %f,%f 0 0 1 %f,%f  l 0,%f  a 4,4 0 0 0 4,4 Z</Geometry>";
+    
+    WCHAR strOut[1024];
+    StringCchPrintf(strOut, ARRAYSIZE(strOut), data,
+        height - 1,
+        leftCorner, leftCorner, leftCorner, leftCorner, leftCorner,
+        ActualWidth() - (leftCorner + rightCorner),
```

Comment:
What if ActualWidth is real small, like smaller than 8?  Any issues there?

Reply:
It won't look right, that's for sure. The MinWidth of a TabViewItem is set to 100, though; it could be changed but I wouldn't expect tabs that small.

Reply:
OK - MinWidth is not likely to be 8 or smaller.